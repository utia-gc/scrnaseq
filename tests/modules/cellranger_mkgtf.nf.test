nextflow_process {

    name "Test Process cellranger_mkgtf"
    script "modules/cellranger_mkgtf.nf"
    process "cellranger_mkgtf"

    test("cellranger_mkgtf filters protein coding genes -- RefSeq GTF.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                // instantiate instances of helper classes
                def refs = new RefsRefSeq()

                """
                input[0] = file("${refs.annotations.gtf.decomp}")
                """
            }
        }

        then {
            // instantiate instances of helper classes
            def refs = new RefsRefSeq()

            // test process status
            assert process.success

            // test filtered GTF output
            with(process.out.gtf_filtered) {
                assert size() == 1
                assert get(0) ==~ /^.*\/${refs.name}_filtered\.gtf/
            }

            // test process reproducibility
            assert snapshot(process.out).match()
        }

    }

}
