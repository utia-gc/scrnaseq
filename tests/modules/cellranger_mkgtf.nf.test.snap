{
    "cellranger_mkgtf filters protein coding genes -- RefSeq GTF.": {
        "content": [
            {
                "0": [
                    "GRCh38.p14_genomic_NC_000021.9_filtered.gtf:md5,08dbcf02adf10b81d9809bf8210a601f"
                ],
                "gtf_filtered": [
                    "GRCh38.p14_genomic_NC_000021.9_filtered.gtf:md5,08dbcf02adf10b81d9809bf8210a601f"
                ]
            }
        ],
        "timestamp": "2023-08-23T16:32:12+0000"
    }
}